lib/puppet_x/puppetlabs/dsc_type_helpers.rb

```diff
@@ -126,6 +126,24 @@ def self.validate_mof_type(mof_type, embeddedinstance_name, name, value)
           end
         end
       end
+
+      def self.reboot_relationship(resource)
+        reboot_resource = resource.catalog.resource(:reboot, 'dsc_reboot')
+        return [] if !reboot_resource
+
+        # if an edge already exists from Reboot[dsc_reboot] to this resource, we have problems
+        if resource.catalog.relationship_graph.edge?(reboot_resource, resource)
+          # TODO: should this be a hard fail? probably?
```

Comment:
This is designed so that multiple DSC resources may direct an edge at the Reboot resource already.  The `TODO` question revolves around whether the user has incorrectly created an edge in the opposite direction.